File: taller/JCZhang-Taller/JCZhang-Taller-Java/src/main/java/models/Chapista.java
package models;

public class Chapista extends Trabajador implements iChapista{

    private final int aniosExp;
    private final String nombre;
    private final int horasDiarias;

    public Chapista(String nombre, int aniosExp, int horasDiarias) {
        super(nombre, aniosExp, horasDiarias);
        this.nombre = nombre;
        this.aniosExp = aniosExp;
        this.horasDiarias = horasDiarias;
    }


    @Override
    public String toString() {
        return "Chapista (" +nombre+" con " + aniosExp +"años de experiencia trabaja" + horasDiarias + "horas al dia)";

    }

    @Override
    public void arreglarChapa(){
        System.out.println("Chapista: Estoy arreglando chapa");

    }

}

File: taller/JCZhang-Taller/JCZhang-Taller-Kotlin/src/main/kotlin/models/Chapista.kt
package models


class Chapista(nombre: String, aniosExp: Int, horasDiaras: Int) : Trabajador(nombre, aniosExp, horasDiaras), iChapista {


    fun Chapista(nombre: String, aniosExp: Int, horasDiaras: Int) {
        this.nombre = nombre
        this.aniosExp = aniosExp
        this.horasDiarias = horasDiaras
    }

    override fun comer() {
        println("Chapista: Hola! soy $nombre y estoy comiendo")
    }

    override fun arreglarChapa(){
        println("Chapista: Estoy arreglando chapa")

    }
    override fun toString(): String {
        return "Chapista ($nombre con $aniosExp años de experiencia trabaja $horasDiarias horas al dia)"
    }

}